doc/schema/schema.md

```diff
@@ -40,10 +41,20 @@ Currently there are 2 beta implementations of this design:
  <li><a class="reference internal" href="#examples">Examples</a></li>
 </ol>
 
+## Terminology
+The terminology used in this specification is based on [xarray's terminology](http://xarray.pydata.org/en/stable/terminology.html), however, no xarray knowledge is assumed in this description. There are also some extensions particular to  the InferenceData case.
+
+* **Variable**: NetCDF-like variables are multidimensional labeled arrays representing a single quantity. Variables and their dimensions must be named. They can also have attributes describing it. Relevant terms related to InferenceData variables are: *variable_name*, *values* (its data), *dimensions*, *coordinates*, and *attributes*.
+* **Dimension**: The dimensions of an object are its named axes. A variable containing 3D data can have dimensions `[chain, draw, dim0]`, thus, its `0th`-dimension is `chain`, its `1st`-dimension is `draw` and so on. Every dimension present in an InferenceData variable must share names with a *coordinate*. Given that dimensions must be named, dimension and dimension name are used equivalents.
+* **Coordinate**: A named array that labels a dimension. A coordinate named `chain` with values `[0, 1, 2, 3]` would label the `chain` dimension. Coordinate names and values can be loosely though of as labels and tick labels along a dimension. 
+* **Group**: Dataset containing one or several variables with a conceptual link between them. Variables inside a group will generally share some dimensions too.
+* **Matching samples**: Two variables (or groups) whose samples match are the ones generated with the same set of samples. Therefore, they will share dimensions and coordinates relative to samples. Sample dimensions (generally `(chain, draw)`) are the ones introduced by the sampling process. 
```

Comment:
nit: should be just "are ones generated" not "the ones generated".

It's not really clear what "relative to samples" means.  "Therefore, they will share the dimensions and coordinates  corresponding to the sampling process."

Maybe an example, showing two variables that have different shapes but that match because they come from the same sampling process?

I'm not sure how *groups* in the same `InferenceData` object would match.  E.g., even in posterior_predictive sampling, PyMC3 might make just a number of draws corresponding to the number of draws times the number of chains in the posterior, so that the posterior and posterior predictive would not match.